src/states/elder/mod.rs

```diff
@@ -616,6 +639,25 @@ impl Elder {
         Ok(())
     }
 
+    fn add_to_proving_section_cache(
+        &mut self,
+        proving_secs: Vec<ProvingSection>,
+        sec_info: SectionInfo,
+    ) {
+        if self.chain.is_new_neighbour(&sec_info) {
+            self.set_of_proving_sections
+                .push_back((proving_secs, sec_info));
+        }
+        while self.set_of_proving_sections.len() > PROVING_SECTION_CACHE_SIZE {
+            let _ = self.set_of_proving_sections.pop_front();
+        }
+    }
+
+    fn remove_from_proving_section_cache(&mut self, sec_info: &SectionInfo) {
```

Comment:
this also depends on if we only choose to cache if leaf sec_info is interesting or any msg going through us.